File: day1.py


def get_input():
  with open("day1.txt", "r") as tf:
    lines = [int(x) for x in tf.read().split()]
  return lines

def part1():
  lines = get_input()
  last = lines[0]
  increases = 0
  for depth in lines:
    if depth > last:
      increases = increases + 1
    last = depth
  #print(increases)

def part2():
  lines = get_input()
  one = [lines[0], lines[1], lines[2]]
  two = [lines[1], lines[2], lines[3]]
  three = [lines[2], lines[3]]
  four = [lines[3]]
  lines = lines[4:]
  current_depth = sum(two)
  new_depth = sum(one)
  one.clear()
  increases = 0
  for depth in lines:
    
    if len(one) == 3:
      new_depth = sum(one)
      one.clear()
    else:
      one.append(depth)
    if len(two) == 3:
      new_depth = sum(two)
      two.clear()
    else:
      two.append(depth)
    if len(three) == 3:
      new_depth = sum(three)
      three.clear()
    else:
      three.append(depth)
    if len(four) == 3:
      new_depth = sum(four)
      four.clear()
    else:
      four.append(depth)
    if new_depth > current_depth:
      increases = increases + 1
    current_depth = new_depth


  if len(one) == 3:
    new_depth = sum(one)
  if len(two) == 3:
    new_depth = sum(two)
  if len(three) == 3:
    new_depth = sum(three)
  if len(four) == 3:
    new_depth = sum(four)
  if current_depth < new_depth:
      increases = increases + 1
  print(increases)


#part1()
part2()

File: day1.rs

use std::fs;

fn main(){
  println!("{}", day1());
  println!("{}", day2());
}

fn day1() -> i32 {
  let depths = get_input();
  let mut prev = depths[0];
  let mut cnt = 0;
  for depth in depths {
    if depth > prev{
      cnt += 1;
    }
    prev = depth;
  }
  return cnt
}

fn day2() -> usize {
  let depths = &get_input();
  depths.windows(4).filter(|w| w.first() < w.last()).count() 
}


fn get_input() -> Vec<i32> {
  let contents = fs::read_to_string("day1.txt")
    .expect("Something went wrong reading file");
  let numbers: Vec<i32> = contents.split("\n")
    .map(|s| s.parse().unwrap())
    .collect();
  return numbers;
}
